modules/react/checkbox/lib/CheckboxInput.tsx

```diff
@@ -0,0 +1,274 @@
+      {
+        '& ~ div:first-of-type': {
+          ...focusRing({
+            width: 2,
+            separation: 2,
+            animate: false,
+            outerColor: cssVar(brand.common.focusOutline),
+          }),
+          borderColor: brand.primary.base,
+          borderWidth: px2rem(2),
+          span: {
+            marginLeft: calc.negate(px2rem(7)),
+          },
+        },
+      },
+  },
+  modifiers: {
+    variant: {
+      inverse: {
+        '& ~ span:first-of-type': {
+          opacity: 0.4,
+        },
+        '& ~ div:first-of-type': {
+          borderColor: base.soap300,
+        },
+
+        // Disabled State for inverse variant
+        '&:not(:where(:checked, :indeterminate, :disabled, :focus-visible, .focus)):where(:hover, .hover, :active, .active) ~ div:first-of-type':
+          {
+            borderColor: base.soap300,
+          },
+        '&:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: base.soap300,
+          backgroundColor: base.frenchVanilla100,
+        },
+        '&:disabled ~ div:first-of-type': {
+          backgroundColor: base.soap300,
+          opacity: system.opacity.disabled,
+        },
+        '&:disabled:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: base.soap300,
+          backgroundColor: base.soap300,
+        },
+
+        // Focus state for inverse variant
+        '&:where(:focus-visible, .focus) ~ div:first-of-type': {
+          borderColor: base.blackPepper400,
+          ...focusRing({
+            width: 2,
+            separation: 0,
+            animate: false,
+            innerColor: cssVar(base.blackPepper400),
+            outerColor: cssVar(base.frenchVanilla100),
+          }),
+        },
+        '&:checked:focus-visible, &:checked.focus, &:indeterminate:focus-visible, &:indeterminate.focus':
+          {
+            '& ~ div:first-of-type': {
+              ...focusRing({
+                width: 2,
+                separation: 2,
+                animate: false,
+                innerColor: cssVar(base.blackPepper400),
+                outerColor: cssVar(base.frenchVanilla100),
+              }),
+              borderColor: base.frenchVanilla100,
+            },
+          },
+      },
+    },
+    disabled: {
+      true: {
+        '&:where(:hover, .hover) ~ span:first-of-type': {
+          boxShadow: 'none',
+        },
+      },
+    },
+    error: {
+      error: {
+        '&:not(:where(:focus-visible, .focus)) ~ div:first-of-type': {
+          borderColor: backgroundVars.inner,
+          boxShadow: `0 0 0 1px ${backgroundVars.inner}, 0 0 0 2px ${backgroundVars.outer}`,
+        },
+        '&:where(:checked, :indeterminate) ~ div:first-of-type': {
+          borderColor: 'transparent',
+          boxShadow: `
+          0 0 0 2px ${base.frenchVanilla100},
+          0 0 0 4px ${backgroundVars.inner},
+          0 0 0 5px ${backgroundVars.outer}`,
+        },
```

Comment:
Question (not RC): I wonder if some of these shared styles should be consolidated?

Reply:
```suggestion
          boxShadow: `
          0 0 0 ${px2rem(2)} ${base.frenchVanilla100},
          0 0 0 ${px2rem(4)} ${backgroundVars.inner},
          0 0 0 ${px2rem(5)} ${backgroundVars.outer}`,
        },
```